iotedgedev/cli.py

```diff
@@ -275,11 +275,36 @@ def deploy(manifest_file):
               show_default=True,
               required=False,
               help="Specify the deployment target condition")
+@click.option("--device_tag",
+              "-dt",
+              default=envvars.get_envvar("DEVICE_TAGS"),
+              required=False,
+              help="Specify the tags to be added to the device twin")
 @with_telemetry
-def iothub_deploy(manifest_file, name, priority, target_condition):
+def iothub_deploy(manifest_file, name, priority, target_condition, device_tag):
     ensure_azure_cli_iot_ext()
     iothub = IoTHub(envvars, output, None, azure_cli)
     iothub.deploy(manifest_file, name, priority, target_condition)
+    if device_tag:
+        edge = Edge(envvars, output, azure_cli)
+        edge.tag(device_tag)
+
+@solution.command(context_settings=CONTEXT_SETTINGS, help="Adds tags to device twin for layered deployments")
+@click.option("--tags",
+              "-t",
+              "tags",
+              required=False,
+              default=envvars.get_envvar("DEVICE_TAGS"),
+              show_default=True,
+              help="Specify the tags to be added to the device twin ")
+@with_telemetry
+def tag(tags):
+    ensure_azure_cli_iot_ext()
+    edge = Edge(envvars, output, azure_cli)
+    edge.tag(tags)
+
+
+main.add_command(tag)
```

Comment:
_Note:_ For consistency reasons I added this as a main command as well. However shouldn't this (as well as the other solution commands) be only subcommands and not main commands?

Reply:
agree with your suggestion, let's keep the new command only as a subscommand